src/components/SmartContractJsonSchema/index.tsx

```diff
@@ -113,9 +113,13 @@ export const JsonSchemaForm = ({
     setSubmitTxError("");
     resetPrepareTx();
 
+    const updatedTxnParams = {
+      ...txnParams,
+      memo: {}, // starting with p23, soroban tx no longer support memos
+    };
```

Comment:
Instead of removing the memo behind the scenes, we could display a message if it's been removed. We can check that when a Soroban operation is selected. Also, we can disable memo picker for Soroban operations.